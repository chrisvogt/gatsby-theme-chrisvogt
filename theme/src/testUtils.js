import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { ThemeProvider } from 'theme-ui'
import theme from './gatsby-plugin-theme-ui'

import store from './store'

export { store }

// React module that provides necessary context to components being tested.
export const TestProvider = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

export const TestProviderWithState = ({ children }) => (
  <ReduxProvider store={store}>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </ReduxProvider>
)
