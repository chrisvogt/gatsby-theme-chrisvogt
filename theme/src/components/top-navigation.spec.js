import React from 'react'
import renderer from 'react-test-renderer'

import { TestProvider } from '../testUtils'
import TopNavigation from './top-navigation'
import useNavigationData from '../hooks/use-navigation-data'
import useSiteMetadata from '../hooks/use-site-metadata'

const mockSiteMetadata = {
  title: 'My Personal Blog & Portfolio'
}

const mockNavigationData = {
  header: {
    left: [
      {
        path: '/about-me',
        slug: 'about-me',
        text: 'About Me',
        title: 'About Me'
      }
    ]
  }
}

jest.mock('../hooks/use-navigation-data')
jest.mock('../hooks/use-site-metadata')

describe('TopNavigation', () => {
  useNavigationData.mockImplementation(() => mockNavigationData)
  useSiteMetadata.mockImplementation(() => mockSiteMetadata)
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <TestProvider>
          <TopNavigation />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
