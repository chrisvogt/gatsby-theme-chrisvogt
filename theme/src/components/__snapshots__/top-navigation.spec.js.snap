// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavigation matches the snapshot 1`] = `
<div
  className="css-zwfbmq-TopNavigation"
>
  <div
    className="css-sl4bvk-TopNavigation"
  >
    <div
      className="css-fjtwc7-TopNavigation"
    >
      <a
        aria-current="page"
        className="css-tmfr8g-TopNavigation"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={Object {}}
      >
        My Personal Blog & Portfolio
      </a>
      <a
        className="css-5hi8j7-TopNavigation"
        href="/about-me"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="About Me"
      >
        About Me
      </a>
    </div>
    <button
      aria-hidden="true"
      className={null}
      onClick={[Function]}
      style={
        Object {
          "MozAppearance": "none",
          "WebkitAppearance": "none",
          "appearance": "none",
          "backgroundColor": "transparent",
          "border": "none",
          "cursor": "pointer",
          "height": "37.599999999999994px",
          "overflow": "hidden",
          "padding": 0,
          "width": "80px",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "height": "132px",
            "justifyContent": "center",
            "marginLeft": "-25.6px",
            "marginTop": "-46px",
            "width": "132px",
          }
        }
      >
        <div />
      </div>
    </button>
  </div>
</div>
`;
